geo/src/algorithm/line_measures/metric_spaces/euclidean.rs

```diff
@@ -0,0 +1,71 @@
+use super::super::Distance;
+use crate::{GeoFloat, Point};
+
+/// Euclidean space measures distance with the pythagorean formula - what you'd measure with a ruler.
+///
+/// You must use projected coordinates with Euclidean space —
+/// for lon/lat points, use the [`Haversine`], [`Geodesic`], or other [metric spaces].
+///
+/// [`Haversine`]: super::Haversine
+/// [`Geodesic`]: super::Geodesic
+/// [metric spaces]: super
+pub struct Euclidean;
+
+/// Calculate the Euclidean distance (a.k.a. pythagorean distance) between two Points
+impl<F: GeoFloat> Distance<F, Point<F>, Point<F>> for Euclidean {
+    /// Calculate the Euclidean distance (a.k.a. pythagorean distance) between two Points
+    ///
+    /// # Units
+    /// - `origin`, `destination`: Point where the units of x/y represent non-angular units
+    ///    — e.g. meters or miles, not lon/lat. For lon/lat points, use the
```

Comment:
In a similar vein, I don't think we should be referring to "meters or miles" here: non-Euclidean distances are often measured in meters / miles, so this is confusing. "Not lon/lat" is OK, but I'd prefer that this is either implicit (you can't measure lon / lat distances in Euclidean space, and we are in Euclidean space, therefore…) or that we decline to explain with physical-world examples completely.

Reply:
> non-Euclidean distances are often measured in meters / miles, so this is confusing.

I believe you, but I honestly didn't know that was true. Can you give me an example?

> you can't measure lon / lat distances in Euclidean space

Oh but you can! And it's a common error that gives you a meaningless result. Look no further than our own example code on the legacy euclidean distance:

```
let p1 = point!(x: -72.1235, y: 42.3521);
let p2 = point!(x: -72.1260, y: 42.45);
let distance = p1.euclidean_distance(&p2);

assert_relative_eq!(distance, 0.09793191512474639);
```

I think it's important to keep simple relatable language and examples to help mitigate that.

Reply:
> I believe you, but I honestly didn't know that was true. Can you give me an example?

Any physical measurement of distance on the earth's surface (or a simplified abstraction of it) is definitionally non-Euclidean because it's a curved surface.

Reply:
Ah I see, that makes sense. Thank you for bearing with me. I don't have a formal GIS background, so a lot of things that may seem obvious, I truly don't know.

I've included the link to [Euclidean Plane], which I hope addresses your ambiguity concern.

I've also linked to the `Transform` trait in reference to what we mean by "projection". While also not 100% precise, I think that it will be helpful to those who know a little, without confusing those who know a lot.

I still think it's important that our documentation aim to help people avoid the most likely errors by saying something like:

> this method takes lng/lat

But the negation:

>  this method does __not__ take lng/lat

Feels awkward and necessarily mysterious without an example. I now understand how Euclidean space is not the only space that could use meters as a coordinate, but *as an example* of what we mean by "not lng/lat",  I think it is clarifying, and unlikely to confuse.

So if I've said something that is incorrect, let me know and I'll try again to fix it. But if it's more so that you think it reads like it was written for dummies, well then I'd please beg of you to have mercy on us dummies.